File: typescript-examples/auth-with-email-otp/utils/typesAndSchemas.ts
import { z } from "zod";

export const createAuthSessionParams = z.object({
  username: z.string().email("Valid email is required"),
  password: z.string().min(8, "Password must be at least 8 characters long"),
});

export interface Contract {
  id: string;
  name: string;
  supplierName: string;
  supplierPhoneNumber: string;
  effectiveDate: string;
  expirationDate: string;
  state: string;
  detailsUrl: string;
}

export const createEmailOTPSessionParams = z.object({
  email: z.string().email(),
});

File: python-examples/auth-with-email-otp/utils/types_and_schemas.py
from pydantic import BaseModel, EmailStr


class CreateAuthSessionParams(BaseModel):
    username: EmailStr
    password: str


class Contract(BaseModel):
    id: str
    name: str
    supplier_name: str
    supplier_phone_number: str
    effective_date: str
    expiration_date: str
    state: str
    details_url: str
